graylog-provider/src/main/java/com/busybusy/graylog_provider/EventJsonizer.kt

```diff
@@ -0,0 +1,211 @@
+        } else {
+            append("\"level\": 6") // default to Informational syslog level
+        }
+    }
+
+    /**
+     * Ensures that user-provided fields conform to the Graylog spec (prefixed by an underscore)
+     */
+    private fun underscorePrefix(fieldName: String): String =
+            if (fieldName.startsWith("_")) "" else "_"
+
+    @Suppress("UNCHECKED_CAST")
+    internal fun getJsonFromMapRecursive(attributeMap: Map<String, Any>): String = buildString {
+        append("{")
+        for (innerAttribute in attributeMap.keys) {
+            val innerAttributeValue = attributeMap[innerAttribute]
+            if (innerAttributeValue is Map<*, *>) // recursive case
+            {
+                // recurse
+                val innerMap = attributeMap[innerAttribute] as Map<String, Any>?
+                append("\"").append(innerAttribute).append("\": ")
+                        .append(getJsonFromMapRecursive(
+                                innerMap!!)).append(", ")
+            } else if (innerAttributeValue is List<*>) // recursive case
+            {
+                append("\"").append(innerAttribute).append("\": ")
+                        .append(getJsonFromListRecursive(
+                                innerAttributeValue)).append(", ")
+            } else {
+                if (innerAttributeValue is String) // gotta use escape quotes for JSON strings
+                {
+                    append("\"").append(innerAttribute).append("\": \"")
+                            .append(getSafeSizeString(
+                                    innerAttributeValue)).append("\", ")
+                } else if (innerAttributeValue is Number || innerAttributeValue is Boolean) {
+                    append("\"").append(innerAttribute).append("\": ")
+                            .append(innerAttributeValue).append(", ")
+                } else {
+                    throw UnsupportedOperationException("Unsupported type for GELF message: " + innerAttributeValue!!.javaClass.simpleName)
+                }
+            }
+        }
+        if (this.toString().length > 1) {
+            deleteCharAt(this.toString().length - 1)
+            deleteCharAt(this.toString().length - 1)
+        }
+        // No else needed: the map was empty, no need to trim the last comma and space
+        append("}")
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    internal fun getJsonFromListRecursive(attributeList: List<*>): String = buildString {
+        append("[")
+        for (element in attributeList) {
+            if (element is Map<*, *>) // recursive case
+            {
+                append(getJsonFromMapRecursive(element as Map<String, Any>)).append(", ")
+            } else if (element is List<*>) // recursive case
+            {
+                append(getJsonFromListRecursive(element)).append(", ")
+            } else {
+                if (element is String) // gotta use escape quotes for JSON strings
```

Comment:
would it work to pull this nested if/else if/else statement out to be on the same level?

Reply:
Yes! In fact, when we put them all on the same level we can replace the whole thing with a nice `when`!